packages/react/src/components/Animation/Animation.tsx

```diff
@@ -102,23 +134,65 @@ class Animation extends UIComponent<WithAsProp<AnimationProps>, any> {
     keyframeParams: PropTypes.object,
     playState: PropTypes.string,
     timingFunction: PropTypes.string,
+    visible: PropTypes.bool,
+    mountOnEnter: PropTypes.bool,
+    appear: PropTypes.bool,
+    unmountOnExit: PropTypes.bool,
+    timeout: PropTypes.oneOfType([
+      PropTypes.number,
+      PropTypes.shape({
+        appear: PropTypes.number,
+        enter: PropTypes.number,
+        exit: PropTypes.number,
+      }),
+    ]),
+  }
+
+  static defaultProps = {
+    timeout: 0,
   }
 
   renderComponent({ ElementType, classes, unhandledProps }) {
-    const { children } = this.props
+    const {
+      children,
+      mountOnEnter,
+      unmountOnExit,
+      timeout,
+      appear,
+      visible,
+      onEnter,
+      onEntering,
+      onEntered,
+      onExit,
+      onExited,
+      onExiting,
+    } = this.props
 
     const child =
       childrenExist(children) && (React.Children.only(children) as React.ReactElement<any>)
     const result = child
       ? React.cloneElement(child, {
-          className: cx(child.props.className, classes.children),
+          ...unhandledProps,
+          className: cx(classes.root, child.props.className),
         })
       : ''
 
     return (
-      <ElementType className={classes.root} {...unhandledProps}>
-        {result}
-      </ElementType>
+      <Transition
+        in={visible}
+        appear={appear}
+        mountOnEnter={mountOnEnter}
+        unmountOnExit={unmountOnExit}
+        timeout={timeout}
+        onEnter={onEnter}
+        onEntering={onEntering}
+        onEntered={onEntered}
+        onExit={onExit}
+        onExiting={onExiting}
+        onExited={onExited}
+      >
+        {state => result}
```

Comment:
https://github.com/reactjs/react-transition-group/blob/318db104c7ba4f279392111c11397d2f053594e7/src/Transition.js#L365-L367

nit: As I see we can do something like this:

```jsx
return (
  <Transition
    {...unhandledProps}
    className={cx(classes.root, child.props.className)}
 >
    {children}
 </Transition>
)
```